### PR TITLE
docs(testing): tech edits

### PR DESCRIPTION
**Pull request by kapunahelewong**
These are copy related tech edits.

### CODE REVIEW

File: public/docs/ts/latest/guide/testing.jade [hunk 1]
```diff
@@ -1871,14 +1967,14 @@ table
...
+        After being called, the `TestBed` configuration is frozen for the duration of the current spec.
 
   tr
     td(style="vertical-align: top") <code>createComponent<T></code>
     td
       :marked
-        Create an instance of a component of type `T` based on the current `TestBed` configuration.
-        Once called, the `TestBed` configuration is frozen for the duration of the current spec.
+        Create a type `T` instance of a component based on the current `TestBed` configuration.
+        After being called, the `TestBed` configuration is frozen for the duration of the current spec.
```
Comment (wardbell) [inline public/docs/ts/latest/guide/testing.jade]: >After calling `createComponent`, the ...

File: public/docs/ts/latest/guide/testing.jade [hunk 2]
```diff
@@ -1871,14 +1967,14 @@ table
         or `styleUrls` because fetching component template and style files is necessarily asynchronous.
         See [above](#compile-components).
 
-        Once called, the `TestBed` configuration is frozen for the duration of the current spec.
+        After being called, the `TestBed` configuration is frozen for the duration of the current spec.
```
Comment (wardbell) [inline public/docs/ts/latest/guide/testing.jade]: >After calling `compileComponents`, the

File: public/docs/ts/latest/guide/testing.jade [hunk 3]
```diff
@@ -1914,14 +2010,15 @@ table
...
+
         What if the service is optional?
 
         The `TestBed.get` method takes an optional second parameter, 
         the object to return if Angular can't find the provider
         (`null` in this example):
       +makeExample('testing/ts/src/app/bag/bag.spec.ts', 'testbed-get')(format=".")
       :marked
-        Once called, the `TestBed` configuration is frozen for the duration of the current spec.
+        After being called, the `TestBed` configuration is frozen for the duration of the current spec.
```
Comment (wardbell) [inline public/docs/ts/latest/guide/testing.jade]: >After calling `get`, the

File: public/docs/ts/latest/guide/testing.jade [hunk 4]
```diff
@@ -1934,8 +2031,8 @@ table
         The testing shims (`karma-test-shim`, `browser-test-shim`) call it for you
         so there is rarely a reason for you to call it yourself.
 
-        This method may be called _exactly once_. Call `resetTestEnvironment` first
-        if you absolutely need to change this default in the middle of your test run.
+        You may call this method _exactly once_. If you absolutely need to change
```
Comment (wardbell) [inline public/docs/ts/latest/guide/testing.jade]: >If you must change ...

File: public/docs/ts/latest/guide/testing.jade [hunk 5]
```diff
@@ -2018,27 +2115,27 @@ table
...
+        When your test code modifies component property values directly, 
+        you probably still have to call `fixture.detectChanges` to trigger data binding updates.
 
-        The default is `false` and testers who prefer fine control over test behavior
+        The default is `false`. Testers who prefer fine control over test behavior
         tend to keep it `false`. 
 
-        Calls `detectChanges` immediately which detects existing changes
-        and will trigger `ngOnInit` if the component has not yet been initialized.
+        Calls `detectChanges` immediately, which detects existing changes
```
Comment (wardbell) [inline public/docs/ts/latest/guide/testing.jade]: Delete this sentence. I moved the sense of it up above.

File: public/docs/ts/latest/guide/testing.jade [hunk 6]
```diff
@@ -2018,27 +2115,27 @@ table
...
     td(style="vertical-align: top") <code>autoDetectChanges</code>
     td
       :marked
-        Set whether the fixture should try to detect changes automatically.
+        Set this to `true` when you want the fixture to detect changes automatically.
 
-        When autodetect is true, the test fixture listens for _zone_ events and calls `detectChanges`. 
-        You probably still have to call `fixture.detectChanges` to trigger data binding updates 
-        when your test code modifies component property values directly.
+        When autodetect is `true`, the test fixture listens for _zone_ events and calls `detectChanges`.
```
Comment (wardbell) [inline public/docs/ts/latest/guide/testing.jade]: >When `autodetect` is `true`, the test fixture calls `detectChanges` immediately after creating the component. Then it listens for pertinent _zone_ events and calls `detectChanges` accordingly.

File: public/docs/ts/latest/guide/testing.jade [hunk 7]
```diff
@@ -2247,21 +2344,16 @@ a(href="#top").to-top Back to top
 
 a#setup-files
 :marked
-  ## Setup files
-  Many think writing tests is fun.
```
Comment (wardbell) [inline public/docs/ts/latest/guide/testing.jade]: One of my favorite introductions _ever_. Sorry to see it go. But go it must.

File: public/docs/ts/latest/guide/testing.jade [hunk 8]
```diff
@@ -2295,7 +2387,7 @@ table(width="100%")
         [SystemJS](https://github.com/systemjs/systemjs/blob/master/README.md) 
         loads the application and test files.
         This script tells SystemJS where to find those files and how to load them.
-        It's the same version of `systemjs.config.js` used by Setup-based applications.
+        It's the same version of `systemjs.config.js` used by setup-based applications.
```
Comment (wardbell) [inline public/docs/ts/latest/guide/testing.jade]: It's the same version of `systemjs.config.js` you installed during [setup](setup.html).

File: public/docs/ts/latest/guide/testing.jade [hunk 9]
```diff
@@ -2367,9 +2459,9 @@ a(href="#top").to-top Back to top
   ### Why put specs next to the things they test?
 
   We recommend putting unit test spec files in the same folder 
-  as the application source code files that they test because
-  - Such tests are easy to find
-  - You see at a glance if a part of our application lacks tests.
+  as the application source code files that they test because:
```
Comment (wardbell) [inline public/docs/ts/latest/guide/testing.jade]: >that they test:

Drop "because"

File: public/docs/ts/latest/guide/testing.jade [hunk 10]
```diff
@@ -8,78 +8,93 @@ block includes
...
+    - [The _tick_ function](#tick)
     - [_jasmine.done_](#jasmine-done)
   * [Test a component with inputs and outputs](#component-with-input-output)
+    - [Test _DashboardHeroComponent_ stand-alone](#dashboard-standalone)
     - [_triggerEventHandler_](#trigger-event-handler)
   * [Test a component inside a test host component](#component-inside-test-host)
   <br><br>  
   * [Test a routed component](#routed-component)
-    - [The _inject_ helper](#inject)
+    - [The _inject_ function](#inject)
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade [hunk 11]
```diff
@@ -164,7 +181,7 @@ table(width="100%")
         End-to-end tests explore the application _as users experience it_.
         In e2e testing, one process runs the real application
         and a second process runs protractor tests that simulate user behavior 
-        and assert that the application responds in the browser as expected.
+        and assert that the application respond in the browser as expected.
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: It should only have an "s" in the indicative though, not in the subjunctive. The subjunctive here is required by "assert", which introduces the subordinate clause of expectation: "that the application respond" and the change of subject. It is the same construction as "She insists that _he see_ a doctor." An alternative in English, which usually obscures the subjunctive, is "She wants _him to see_ a doctor." If the first clause and second clause have different subjects and there is doubt, disbelief, denial, or imposition of will (under which expectation falls), the second clause must use the subjunctive. So you get things like: 

> Mom expects that he eat his lunch.
> The law requires that the school have a reading program.

@wardbell I hope you read that in Leonard Nimoy's voice.

File: public/docs/ts/latest/guide/testing.jade [hunk 12]
```diff
@@ -229,30 +246,31 @@ a#1st-karma-test
 :marked
   **Put spec files somewhere within the `src/app/` folder.**
   The `karma.conf.js` tells karma to look for spec files there,
-  for reasons explained [below](#q-spec-file-location).
+  for reasons explained [later](#q-spec-file-location).
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: You made me laugh. I changed it back. Below sounds more natural and lets you know it's in the same page.

File: public/docs/ts/latest/guide/testing.jade [hunk 13]
```diff
@@ -229,30 +246,31 @@ a#1st-karma-test
...
 
   Add the following code to `src/app/1st.spec.ts`.
 +makeExample('testing/ts/src/app/1st.spec.ts', '', 'src/app/1st.spec.ts')(format='.')
 
 a#run-karma
 :marked
-  ### Run karma
-  Compile and run it in karma from the command line with this command:
+  ### Run with karma
+  Compile and run it in `karma` from the command line using the following command:
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done. Thank you for the clarification.

File: public/docs/ts/latest/guide/testing.jade [hunk 14]
```diff
@@ -229,30 +246,31 @@ a#1st-karma-test
...
 code-example(format="." language="bash").
   npm test
 :marked
-  The command compiles the application and test code and starts karma.
+  The command compiles the application and test code and starts `karma`.
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade [hunk 15]
```diff
@@ -283,33 +301,34 @@ code-example(format="." language="bash").
...
-    - Open the browser's “Developer Tools” (F12 or Ctrl-Shift-I).
-    - Pick the "sources" section
-    - Open the `1st.spec.ts` test file (Ctrl-P, then start typing the name of the file).
-    - Set a breakpoint in the test
-    - Refresh the browser … and it stops at the breakpoint.
+  Debug specs in the browser in the same way that you debug an application.
+
+    1. Reveal the `karma` browser window (hidden earlier).
+    1. Click the **DEBUG** button; it opens a new browser tab and re-runs the tests.
+    1. Open the browser's “Developer Tools” (`F12` or `Ctrl-Shift-I`).
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Changed.

File: public/docs/ts/latest/guide/testing.jade [hunk 16]
```diff
@@ -283,33 +301,34 @@ code-example(format="." language="bash").
...
+    1. Pick the "sources" section.
+    1. Open the `1st.spec.ts` test file (Ctrl-P, then start typing the name of the file).
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade [hunk 17]
```diff
@@ -363,8 +383,8 @@ a#testbed
   The metadata lack `imports` because (a) the default testing module configuration already has what `BannerComponent` needs
   and (b) `BannerComponent` doesn't interact with any other components.
 
-  Call `configureTestingModule` within a `BeforeEach` so that,
-  before each spec runs, the `TestBed` can reset itself to a base state.
+  Call `configureTestingModule` within `BeforeEach` so that,
+  before each spec runs, `TestBed` can reset itself to a base state.
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: So much better. Done.

File: public/docs/ts/latest/guide/testing.jade [hunk 18]
```diff
@@ -466,12 +491,14 @@ a#auto-detect-changes
 
   The `BannerComponent` tests frequently call `detectChanges`.
   Some testers prefer that the Angular test environment run change detection automatically.
-  That's possible by configuring the `TestBed` with the `ComponentFixtureAutoDetect` provider which
-  you first import from the testing utility library ...
+
+  That's possible by configuring the `TestBed` with the `ComponentFixtureAutoDetect` provider. 
+  First import from the testing utility library:
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade [hunk 19]
```diff
@@ -466,12 +491,14 @@ a#auto-detect-changes
...
 +makeExample('testing/ts/src/app/banner.component.detect-changes.spec.ts', 'import-ComponentFixtureAutoDetect', 'src/app/banner.component.detect-changes.spec.ts (import)')(format='.')
 :marked
-  ... and then add to the `providers` array of the testing module configuration:
+  Then add to the `providers` array of the testing module configuration:
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: You have a good eye! :)

File: public/docs/ts/latest/guide/testing.jade [hunk 20]
```diff
@@ -666,12 +700,15 @@ a#testbed-get
   :marked
     The [`inject`](#inject)  utility function is another way to get one or more services from the test root injector.
     
-    See the section "[_Override a component's providers_](#component-override)" for a use case
-    in which `inject` and `TestBed.get` do not work and you must get the service from the component's injector.
+    For a use case in which `inject` and `TestBed.get` do not work and
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Much smoother.

File: public/docs/ts/latest/guide/testing.jade [hunk 21]
```diff
@@ -710,12 +747,14 @@ a#component-with-async-service
...
   The `TwainComponent` handles the display, delegating the server request to the `TwainService`.
+
   Both are in the `src/app/shared` folder because the author intends to display Twain quotes on other pages someday.
   Here is the `TwainComponent`.
 +makeExample('testing/ts/src/app/shared/twain.component.ts', 'component', 'src/app/shared/twain.component.ts')(format='.')
+
 :marked
-  The `TwainService` implementation is irrelevant at this point.
-  It is sufficient to see within `ngOnInit` that `twainService.getQuote` returns a promise which means it is asynchronous.
+  At this point, the `TwainService` implementation is irrelevant.
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade [hunk 22]
```diff
@@ -731,7 +770,7 @@ a#service-spy
 +makeExample('testing/ts/src/app/shared/twain.component.spec.ts', 'spy')(format='.')
 :marked
   The spy is designed such that any call to `getQuote` receives an immediately resolved promise with a test quote.
-  The spy bypasses the actual `getQuote` method and therefore will not contact the server.
+  The spy bypasses the actual `getQuote` method and therefore doesn't contact the server.
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Agreed. Done.

File: public/docs/ts/latest/guide/testing.jade [hunk 23]
```diff
@@ -744,22 +783,25 @@ a#service-spy
...
 :marked
   ### Synchronous tests
   The first two tests are synchronous. 
   Thanks to the spy, they verify that `getQuote` is called _after_
   the first change detection cycle during which Angular calls `ngOnInit`. 
 
-  Neither test can prove that a value from the service is be displayed.
-  The quote itself has not arrived, despite the fact that the spy returns a resolved promise.
+  Neither test can prove that a value from the service is displayed.
+  The quote itself hasn't arrived, despite the fact that the spy returns a resolved promise.
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade [hunk 24]
```diff
@@ -818,28 +862,27 @@ a#fake-async
...
   chained in a long sequence of promise callbacks.
 
 a#jasmine-done
 :marked
-  ## _jasmine.done_
+  ### _jasmine.done_
 
-  While `fakeAsync` and even `async` function greatly simplify Angular asynchronous testing,
-  you can still fallback to the traditional Jasmine asynchronous testing technique.
+  While the `fakeAsync` and even the `async` functions greatly simplify Angular asynchronous testing,
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade [hunk 25]
```diff
@@ -848,11 +891,11 @@ a#jasmine-done
...
 :marked
-  Although we have no direct access to the `getQuote` promise inside `TwainComponent`,
-  the spy does and that makes it possible to wait for `getQuote` to finish.
+  Although there is no direct access to the `getQuote` promise inside `TwainComponent`,
+  the spy has direct access, which makes it possible to wait for `getQuote` to finish.
   
-  The `jasmine.done` technique, while discouraged, may become necessary when neither `async` nor `fakeAsync`
-  can tolerate a particular asynchronous activity. That's rare but it happens.
+  The `jasmine.done` technique, while discouraged, may be necessary when neither `async` nor `fakeAsync`
+  can tolerate a particular asynchronous activity, a rare occurrence.
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Replaced.

File: public/docs/ts/latest/guide/testing.jade [hunk 26]
```diff
@@ -862,49 +905,53 @@ a#component-with-input-output
...
+  The host uses a property binding to set the input property and an event binding to
   listen to events raised by the output property.
 
   The testing goal is to verify that such bindings work as expected. 
   The tests should set input values and listen for output events.
 
   The `DashboardHeroComponent` is a tiny example of a component in this role.
   It displays an individual hero provided by the `DashboardComponent`. 
-  Clicking that hero tells the `DashboardComponent` that the user has selected the hero.
+  Clicking that hero informs the `DashboardComponent` that the user has selected the hero.
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done!

File: public/docs/ts/latest/guide/testing.jade [hunk 27]
```diff
@@ -862,49 +905,53 @@ a#component-with-input-output
...
 
   The `DashboardHeroComponent` is embedded in the `DashboardComponent` template like this:
 +makeExample('testing/ts/src/app/dashboard/dashboard.component.html', 'dashboard-hero', 'src/app/dashboard/dashboard.component.html (excerpt)')(format='.')
 :marked
-  The `DashboardHeroComponent` appears in an `*ngFor` repeater which sets each component's `hero` input property
-  to the iteration value and listens for the components `selected` event.
+  The `DashboardHeroComponent` appears in an `*ngFor` repeater, which sets each component's `hero` input property
+  to the iteration value and listens for the component's `selected` event.
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade [hunk 28]
```diff
@@ -862,49 +905,53 @@ a#component-with-input-output
...
   
   Here's the component's definition:
 +makeExample('testing/ts/src/app/dashboard/dashboard-hero.component.ts', 'component', 'src/app/dashboard/dashboard-hero.component.ts (component)')(format='.')
 :marked
   While testing a component this simple has little intrinsic value, it's worth knowing how.
-  Three approaches come to mind:
-  1. Test it as used by `DashboardComponent`
-  1. Test it as a stand-alone component
-  1. Test it as used by a substitute for `DashboardComponent`
+  You can use one of these methods:
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Ah yes. I try to be careful using "methods", but missed this. Thank you!

File: public/docs/ts/latest/guide/testing.jade [hunk 29]
```diff
@@ -915,17 +962,19 @@ a#component-with-input-output
 
   The first test follows:
 +makeExample('testing/ts/src/app/dashboard/dashboard-hero.component.spec.ts', 'name-test', 'src/app/dashboard/dashboard-hero.component.spec.ts (name test)')(format='.')
+
 :marked
   It verifies that the hero name is propagated through to template with a binding.
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade [hunk 30]
```diff
@@ -915,17 +962,19 @@ a#component-with-input-output
...
-  There's a twist. The template passes the hero name through the Angular `UpperCasePipe` so the
+  However, the template passes the hero name through the Angular `UpperCasePipe`, so the
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade [hunk 31]
```diff
@@ -952,28 +1001,33 @@ a#trigger-event-handler
...
-  doesn't care about the event object.
+  The test assumes (correctly in this case) that the runtime 
+  event handler&mdash;the component's `click()` method&mdash;doesn't 
+  care about the event object.
   
-  Other handlers will be less forgiving. 
-  For example, the `RouterLink` directive expects an object with a `button` property indicating the mouse button that was pressed. 
-  The directive throws an error if the event object doesn't do this correctly.
+  Other handlers are less forgiving. For example, the `RouterLink` 
+  directive expects an object with a `button` property, indicating
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade [hunk 32]
```diff
@@ -952,28 +1001,33 @@ a#trigger-event-handler
...
+  custom event object as the second parameter. The default is a (partial) 
   <a href="https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/button" target="_blank">left-button mouse event object</a> 
   accepted by many handlers including the `RouterLink` directive.
 
 .callout.is-critical
   header click() is not an Angular testing utility
   :marked
     The `click()` helper function is **not** one of the Angular testing utilities. 
-    It's a function defined in _this guide's sample code_ and used by all of the sample tests. 
+    It's a function defined in _this guide's sample code_, and which all of the sample tests use.
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade [hunk 33]
```diff
@@ -1199,11 +1256,11 @@ figure.image-display
   But there's already plenty of template complexity.
 +makeExample('testing/ts/src/app/hero/hero-detail.component.html', '', 'src/app/hero/hero-detail.component.html')(format='.')
 :marked
-  To fully exercise the component, the test needs ...
-  * to wait until a `hero` arrives before `*ngIf` allows any element in DOM
-  * element references for the title name span and name input-box to inspect their values
-  * two button references to click
-  * spies on component and router methods
+  To fully exercise the component, the test needs:
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade [hunk 34]
```diff
@@ -1213,15 +1270,19 @@ figure.image-display
 :marked
   Now the important hooks for component manipulation and inspection are neatly organized and accessible from an instance of `Page`.
 
+
   A `createComponent` method creates a `page` and fills in the blanks once the `hero` arrives.
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade [hunk 35]
```diff
@@ -1241,28 +1302,35 @@ a#import-module
...
+
 
   One approach is to configure the testing module from the individual pieces as in this example:
 +makeExample('testing/ts/src/app/hero/hero-detail.component.spec.ts', 'setup-forms-module', 'src/app/hero/hero-detail.component.spec.ts (FormsModule setup)')(format='.')
+
 :marked
-  Because many app components need the `FormsModule` and the `TitleCasePipe`, the developer created 
-  a `SharedModule` to combine these and other frequently requested parts.
-  The test configuration can use the `SharedModule` too as seen in this alternative setup:
+  Another option is a custom module that _you_ make. For example, here, `SharedModule`
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Ah ok. Thank you. Reverted!

File: public/docs/ts/latest/guide/testing.jade [hunk 36]
```diff
@@ -1419,7 +1498,10 @@ a#router-link-stub
   ### Stubbing the _RouterLink_
 
   The `RouterLinkStubDirective` contributes substantively to the test
+
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Removed.

File: public/docs/ts/latest/guide/testing.jade [hunk 37]
```diff
@@ -1419,7 +1498,10 @@ a#router-link-stub
...
 +makeExample('testing/ts/src/testing/router-stubs.ts', 'router-link', 'testing/router-stubs.ts (RouterLinkStubDirective)')(format='.')
+
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Ha, ok. Not sure how I did it (yes ok, probably by hitting enter, haha) but I will be more careful.

File: public/docs/ts/latest/guide/testing.jade [hunk 38]
```diff
@@ -1474,7 +1556,7 @@ a#why-stubbed-routerlink-tests
 
   A _different_ battery of tests can explore whether the application navigates as expected
   in the presence of conditions that influence guards such as whether the user is authenticated and authorized.
-  A future guide update will explain how to write such tests with the `RouterTestingModule`.
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Made it an `.is_helpful` note.

File: public/docs/ts/latest/guide/testing.jade [hunk 39]
```diff
@@ -1535,12 +1618,14 @@ a#attribute-directive
   However, testing a single use case is unlikely to explore the full range of a directive's capabilities.
   Finding and testing all components that use the directive is tedious, brittle, and almost as unlikely to afford full coverage.
 
-  [Isolated unit tests](#isolated-unit-tests) might be helpful. 
-  But attribute directives like this one tend to manipulate the DOM. 
+  [Isolated unit tests](#isolated-unit-tests) might be helpful, 
+  but attribute directives like this one tend to manipulate the DOM. 
   Isolated unit tests don't and therefore don't inspire confidence in the directive's efficacy.
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade [hunk 40]
```diff
@@ -1583,18 +1668,18 @@ a#isolated-unit-tests
...
   with _isolated_  unit tests that don't depend upon Angular.
-  Such tests are often smaller and  easier to read, write and maintain.
+  Such tests are often smaller and  easier to read, write, and maintain.
 
-  They don't
-  * import from the Angular test libraries
-  * configure a module
-  * prepare dependency injection `providers`
-  * call `inject` or `async` or `fakeAsync`
+  They don't:
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade [hunk 41]
```diff
@@ -1583,18 +1668,18 @@ a#isolated-unit-tests
...
+  * Import from the Angular test libraries.
+  * Configure a module.
+  * Prepare dependency injection `providers`.
+  * Call `inject` or `async` or `fakeAsync`.
 
-  They do
-  * exhibit standard, Angular-agnostic testing techniques
-  * create instances directly with `new`
-  * substitute test doubles (stubs, spys, and mocks) for the real dependencies.
+  They do:
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade [hunk 42]
```diff
@@ -1660,20 +1749,24 @@ a#isolated-pipe-tests
   ### Pipes
   Pipes are easy to test without the Angular testing utilities.
 
-  A pipe class has one method, `transform`, that turns an input to an output. 
+  A pipe class has one method, `transform`, that turns an input into an output.
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade [hunk 43]
```diff
@@ -1660,20 +1749,24 @@ a#isolated-pipe-tests
...
+  Here's a naive implementation with a regular expression.
 +makeExample('testing/ts/src/app/shared/title-case.pipe.ts', '', 'src/app/shared/title-case.pipe.ts')(format='.')
 :marked
   Anything that uses a regular expression is worth testing thoroughly.
   Use simple Jasmine to explore the expected cases and the edge cases.
+
 +makeExample('testing/ts/src/app/shared/title-case.pipe.spec.ts', 'excerpt', 'src/app/shared/title-case.pipe.spec.ts')
+
+a#write-tests
+docs(testing): copy edits to end of TOC
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Aw man. Cleaned up.

File: public/docs/ts/latest/guide/testing.jade [hunk 44]
```diff
@@ -1693,10 +1786,11 @@ a#isolated-component-tests
   The following Angular test demonstrates that clicking a button in the template leads
   to an update of the on-screen message.
 +makeExample('testing/ts/src/app/bag/bag.spec.ts', 'ButtonComp', 'src/app/bag/bag.spec.ts (ButtonComp)')(format='.')
+
 :marked
-  The assertions verify the data binding flow from one HTML control (the `<button>`) to the component and 
+  The assertions verify that the data binding flows from one HTML control (the `<button>`) to the component and
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade [hunk 45]
```diff
@@ -1810,13 +1904,13 @@ table
 
 a#testbed-class-summary
 :marked
-  ## _TestBed_ Class Summary
+  ### _TestBed_ class summary
   The `TestBed` class is one of the principal Angular testing utilities. 
-  Its API is quite large and can be overwhelming until you've explored it first
-  a little at a time. Read the early part of this guide first
+  Its API is quite large and can be overwhelming until you've explored it,
+  a little at a time. First, read the early part of this guide
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade (hunk 2 above)
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade [hunk 46]
```diff
@@ -1871,14 +1967,14 @@ table
...
-        Once called, the `TestBed` configuration is frozen for the duration of the current spec.
+        After being called, the `TestBed` configuration is frozen for the duration of the current spec.
 
   tr
     td(style="vertical-align: top") <code>createComponent<T></code>
     td
       :marked
-        Create an instance of a component of type `T` based on the current `TestBed` configuration.
-        Once called, the `TestBed` configuration is frozen for the duration of the current spec.
+        Create a type `T` instance of a component based on the current `TestBed` configuration.
```
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Reverted.

File: public/docs/ts/latest/guide/testing.jade (hunk 1 above)
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade (hunk 3 above)
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade (hunk 4 above)
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade (hunk 6 above)
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade (hunk 5 above)
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade (hunk 7 above)
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Agreed!!!

File: public/docs/ts/latest/guide/testing.jade (hunk 8 above)
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done.

File: public/docs/ts/latest/guide/testing.jade (hunk 9 above)
Comment (kapunahelewong) [inline public/docs/ts/latest/guide/testing.jade]: Done. Also noticed that that sentence started with "We". Fixed that, too.
